buyer-api/src/routes/dataOrders/offchainData.js

```diff
@@ -42,8 +42,7 @@ router.get(
       buyerInfoId,
     } = req.dataOrder;
 
-    const allowedNotaries = await Promise.all(notariesAddresses
-      .map(notaryAddress => notaries.fetch(notaryAddress)));
+    const allowedNotaries = await getNotariesInfo(notariesAddresses);
```

Comment:
I now see that this approach can be problematic. Let me ask: Shouldn't this endpoint ensure the presence of the information from the notaries inside `notariesAdressess`? I mean, what happens if the data order has two notaries, and one of the two requests fails? The response is cached and served like this (in bapi and wapi), and I'm not sure how the app handles this.
Maybe we should let the endpoint fail and let wapi retry, or implement a caching strategy for the notary info. What do you think?

Reply:
That's true, but letting it fail is even more problematic. Whoever registers as a notary with any invalid URL blocks the whole platform immediately.
But I see your point. Maybe we should state that additional information will be there only if the service is available at the time of the request, but the onchain info will always be included. That way there won't be any missing notary, and API consumers know that the rest of the info might not be there if the notary is not online.